visualizer/exampleGenerationRequests.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-env browser */
+
+'use strict';
+
+// TODO: handle invalid grammar in textbox
+
+(function(root, initModule) {
+  if (typeof exports === 'object') {
+    module.exports = initModule;
+  } else {
+    initModule(root.ohm, root.ohmEditor, root.exampleWorkerManager,
+               root.utils, root.domUtil, root.autosize);
+  }
+})(this, function(ohm, ohmEditor, exampleWorkerManager, utils, domUtil, autosize) {
+
+  var neededExamples = [];
+
+  var focusedElement = null;
+  var focusedRuleName = '';
+
+  // TODO: clicking off of textarea triggers refresh, deleting content
+
+  // logic for arrow to open drawer and show needed examples menu
+  var examplesNeededDrawer = domUtil.$('#examplesNeededDrawer');
+  var examplesNeededIndicator = domUtil.$('#examplesNeededDrawer .indicator');
+  var exampleSplitter = domUtil.$('#exampleSplitter');
+  examplesNeededDrawer.addEventListener('click', function() {
+    var ExampleRequestContainer = domUtil.$('#ExampleRequestContainer');
+    ExampleRequestContainer.classList.toggle('hidden');
+    exampleSplitter.classList.toggle('disabled',
+      ExampleRequestContainer.classList.contains('hidden')
+    );
+  });
+
+  var timeout = null;
+  exampleWorkerManager.addListener('received:neededExamples', function(updatedNeededExamples) {
+    if (timeout) {
+      clearTimeout(timeout);
+    }
+
+    timeout = setTimeout(function() {
+      var neededExampleList = domUtil.$('#exampleRequests ul');
+
+      neededExamples = updatedNeededExamples;
+
+      Array.prototype.slice.call(neededExampleList.children)
+        .forEach(function(childNode) {
+          if (childNode.firstChild !== focusedElement) {
+            neededExampleList.removeChild(childNode);
+          }
+        });
+
+      neededExamples.filter(function(ruleName) {
+        return ruleName !== focusedRuleName;
+      }).forEach(function(ruleName) {
+        var li = domUtil.createElement('li');
+        li.appendChild(makeExampleRequest(ruleName));
+        neededExampleList.appendChild(li);
+      });
+
+      if (neededExamples.length === 0) {
```

Comment:
Nit: This could be `examplesNeededIndicator.classList.toggle('active', neededExamples.length !== 0);`